File: back/src/main/java/org/collabtask/user/core/contracts/IUserService.java
package org.collabtask.user.core.contracts;

import org.collabtask.helpers.PaginatedResponse;
import org.collabtask.helpers.Pagination;
import org.collabtask.user.core.dto.CreateUser;
import org.collabtask.user.core.dto.UserClient;
import org.collabtask.user.core.exception.UserNotFoundException;

import io.smallrye.mutiny.Uni;

public interface IUserService {
    Uni<PaginatedResponse<UserClient>> findAll(Pagination pagination);

    Uni<UserClient> findById(String id) throws UserNotFoundException;

    Uni<UserClient> findByEmail(String email) throws UserNotFoundException;

    Uni<UserClient> create(CreateUser data);

    Uni<UserClient> delete(String id) throws UserNotFoundException;
}

File: front/app/user/core/service/IUserService.ts
import type { ResultAsync } from "neverthrow";
import type { CreateUser, PaginatedUser, User } from "../user";

export interface IUserService {
  findAll(): ResultAsync<PaginatedUser, Error>;
  findById(id: string): ResultAsync<User, Error>;
  findByEmail(email: string): ResultAsync<User, Error>;
  create(createUser: CreateUser): ResultAsync<User, Error>;
  deleteById(id: string): ResultAsync<User, Error>;
}
